v2/client/options.go

```diff
@@ -126,3 +126,16 @@ func WithBlockingCallback() Option {
 		return nil
 	}
 }
+
+// WithAckMalformedevents causes malformed events received within StartReceiver to be acknowledged
+// rather than being permanently not-acknowledged. This can be useful when a protocol does not
+// provide a responder implementation and would otherwise cause the receiver to be partially or
+// fully stuck.
+func WithAckMalformedEvent() Option {
+	return func(i interface{}) error {
```

Comment:
I wondered if it would be preferable to modify the pattern here with Uber's suggested implementation of [functional options]. For example:

```go
type Option interface {
  apply(*ceClient) error
}

type ackMalformedEvent bool

func WithAckMalformedEvent() Option {
  return ackMalformedEvent(true)
}

func (a *ackMalformedEvent) apply(c *ceClient) error {
  c.ackMalformedEvent = bool(a)
  return nil
}
```

It's slightly more surface area, but in LOC terms they're about the same. It also means there's no need to cast `interface{}` and they're pretty easy to unit test. I don't think it'd be a breaking change to the API because the `Option` type is not something external packages could meaningfully implement, unless consumers are using it to bundle multiple options together.

If that's something you'd like to see, I can add it in a follow-up PR.

Reply:
IMHO your proposal in this PR (using casting) is OK and the small overhead of casting vs introducing new signatures and potentially breaking APIs is not worth the effort.